hibernate-core/src/main/java/org/hibernate/query/results/dynamic/DynamicFetchBuilderContainer.java

```diff
@@ -13,22 +14,22 @@ public interface DynamicFetchBuilderContainer {
 	/**
 	 * Locate an explicit fetch definition for the named fetchable
 	 */
-	FetchBuilder findFetchBuilder(String fetchableName);
+	FetchBuilder findFetchBuilder(Fetchable fetchable);
```

Comment:
Not sure if we consider this to be "API", but wanted to mention the backwards compatibility concern.

Reply:
I think it's incredibly unlikely that any user code depends on `DynamicFetchBuilderContainer` since there are no SPIs which return or accept this type.

And therefore I don't think there's any need to worry about breaking it *in a new major version*.

Reply:
I agree - https://hibernate.atlassian.net/browse/HHH-18847

Reply:
The work I've done @ https://github.com/hibernate/hibernate-orm/pull/9257 addresses this specific concern.

However, a natural continuation there is that everything else in the `org.hibernate.query.results` package should really be an spi.  I started down this path, but it affects a few things that are in the same boat imo.  E.g. this has ties with the `org.hibernate.query.named`package, which itself ought to be spi imo.  I'm going to follow up with a separate pr that follows that rabbit hole so we can see the full impact.

Reply:
> However, a natural continuation there is that everything else in the org.hibernate.query.results package should really be an spi

Well, unless we forsee actually adding some form of support for an application creating these directly.  A sort of improved approach to `NativeQuery#addReturn`, etc.  That was the original plan, but we've not done it and noone has asked for it so...

Reply:
> @DavideD wrt ^^, as far as I could see this change would have a minimal impact on HR so wanted to get your take. HR has an implementation of the main contract there, but like I said I think the impact for HR is minimal.

Yes, the impact should be minimal.

Reply:
I can try tomorrow to update the code in Hibernate Reactive and make sure that there are not surprises.